design_docs/config-file_reorganization.md

```diff
@@ -0,0 +1,108 @@
+```
+# a list of which analyses to generate.  Valid names are:
+#   'ohc_timeseries', 'sst_timeseries', 'sst_modelvsobs', 'sss_modelvsobs',
+#   'mld_modelvsobs', 'seaice_timeseries', 'seaice_modelvsobs'
+# the following shortcuts exist:
+#   'all' -- all analyses will be run
+#   'all_timeseries' -- all time-series analyses will be run
+#   'all_modelvsobs' -- all analyses comparing model climatologies with
+#                       observations will be run
+#   'all_ocean' -- all ocean analyses will be run
+#   'all_seaice' -- all sea-ice analyses will be run
+#   'no_ohc_timeseries' -- skip the 'ohc_timeseries' (and similarly with the other analyses).
+#   'no_ocean', 'no_timeseries', etc. -- in analogy to 'all_*', skip the given category of analysis
+generate = ['all']
+```
+Where there are conflicts between items in the `generate` list, successive items will override earlier items.  For example, `generate = ['all', 'no_ohc_timeseries']` will generate all analyses except `ohc_timeseries`.  As another example, `generate = ['all', 'no_ocean', 'all_timeseries']` would generate all diagnostics except those comparing ocean model results with observations (and previous model results).  (Note that a more efficient and intuitive way to do the same would be `generate = ['all_seaice', 'all_timeseries']`.)
+
+An analogous approach will also be added at the command line, for example:
+```
+./run_analysis.py config.analysis --generate all,no_ocean,all_timeseries
+```
+(I am open to other syntax.)  If the --generate flag is used on the command line, it will replace the generate option in the config file.
+
+As an aside, I note that it is not clear if future analysis modules will fit neatly into categories like "time series" and "model vs. observations", and these categories are not meant to be all-encompassing.
+
+<h2> Implementation: there should be a simplified template for config files <br>
+Date last modified: 2017/01/29 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+Such a template has been implemented in #86.  A subdirectory `configs` will be added with several examples from runs on LANL IC and on Edison at NERSC.  Other examples can be added as appropriate and useful.
+
+<h2> Implementation: removal of ACME specific config options <br>
+Date last modified: 2017/01/29 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+This item needs some discussion.  in #86, I would like to renamed `ref_casename_v0` to simply `ref_casename`.  `casename` and `ref_casename_v0` are used for file names and figure titles (essentially legends).  It would be useful to discuss what the relevant equivalents would be fore standalone MPAS runs.
+
+<h1> Testing </h1>
+
+<h2> Testing and Validation: a simple way of turning on and off individual analysis modules <br>
+Date last modified: 2017/01/29 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+This will be difficult to test with CI since we currently don't have CI that can perform tests with `run_analysis.py`.
```

Comment:
Coveralls and landscape are github "plug-ins" that can be automatically run, like CI, to provide diagnostics regarding code quality.  They can indicate parts of the code that may need further testing to provide a robust and maintainable code base.  At the present, I'm not suggesting we use them, just keep them in mind if we have needs related to identification of insufficient test in particular.